cmd/lakectl/cmd/ingest.go

```diff
@@ -69,7 +69,14 @@ var ingestCmd = &cobra.Command{
 			path = path + PathDelimiter // append a path delimiter (slash) if not passed by the user, and it's not an empty path in lakeFS
 		}
 		go func() {
-			err := store.Walk(ctx, s3EndpointURL, from, func(e store.ObjectStoreEntry) error {
+			walker, err := store.GetWalker(ctx, store.WalkerOptions{
+				S3EndpointURL: s3EndpointURL,
+				StorageURI:    from,
+			})
+			if err != nil {
+				DieFmt("error creating object-store walker: %v", err)
+			}
+			err = walker.Walk(ctx, store.WalkOptions{}, func(e store.ObjectStoreEntry) error {
```

Comment:
Question - Now that we have a similar server side implementation - is this logic about to move to the server side as well, or is it here to stay?

Reply:
Good question. They provide different capabilities. 
- `lakectl ingest` is more intuitive and stage the entries to the workspace, but doesn't scale to millions of objects.
- New server side import scales better, but has all or nothing approach to create a commit.

I think a possible followup is to consolidate the ingest/import/UI import features into 1/2, but not part of this PR.